pkg/controller/controller_binding.go

```diff
@@ -349,38 +352,70 @@ func isPlanBindable(serviceClass *v1alpha1.ServiceClass, plan *v1alpha1.ServiceP
 
 func (c *controller) injectBinding(binding *v1alpha1.Binding, credentials map[string]interface{}) error {
 	glog.V(5).Infof("Creating Secret %v/%v", binding.Namespace, binding.Spec.SecretName)
-	secret := &v1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      binding.Spec.SecretName,
-			Namespace: binding.Namespace,
-		},
-		Data: make(map[string][]byte),
-	}
 
+	secretData := make(map[string][]byte)
 	for k, v := range credentials {
 		var err error
-		secret.Data[k], err = serialize(v)
+		secretData[k], err = serialize(v)
 		if err != nil {
 			return fmt.Errorf("Unable to serialize credential value %q: %v; %s",
 				k, v, err)
 		}
 	}
 
-	found := false
+	secretClient := c.kubeClient.Core().Secrets(binding.Namespace)
+	// Retry loop for upserting the Secret
+	for {
```

Comment:
1. I don’t see how it could be achieved without broker storing the request payload
2. Broker should also be able to re-fetch the same credentials, which again makes it stateful

I'm pretty sure this requirement is not respected by stateless brokers. Thus, Secret retry loop should improve the UX.
Will create an issue in OSB repo with the question about Binding retries.